Dockerfile

```diff
@@ -127,19 +127,10 @@ RUN set -x \
 # IMPORTANT: If the version of Go is updated, the Windows to Linux CI machines
 #            will need updating, to avoid errors. Ping #docker-maintainers on IRC
 #            with a heads-up.
-ENV GO_VERSION 1.5.4
+ENV GO_VERSION 1.6.2
 RUN curl -fsSL "https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz" \
 	| tar -xzC /usr/local
 
-# !!! TEMPORARY HACK !!!
-# Because of https://github.com/golang/go/issues/15286 we have to revert to Go 1.5.3 for windows/amd64 in master
-# To change which version of Go to compile with, simply prepend PATH with /usr/local/go1.5.3/bin
-# and set GOROOT to /usr/local/go1.5.3
-ENV HACK_GO_VERSION 1.5.3
-RUN curl -fsSL "https://storage.googleapis.com/golang/go${HACK_GO_VERSION}.linux-amd64.tar.gz" \
-	| tar -xzC /tmp \
-	&& mv /tmp/go "/usr/local/go${HACK_GO_VERSION}"
-
```

Comment:
Are we sure this is no longer needed? Following this comment: https://github.com/golang/go/issues/15286#issuecomment-217323815 it looks like there's still a workaround needed (because a fix didn't make it for Go 1.6.2)

Also, if we're reverting the Hack/Workaround, it looks like there's also changes to be made to `hack/make/cross`; see https://github.com/docker/docker/pull/22082

ping @jstarks @jhowardmsft could you have a look?